### ISSUE TITLE
stop_time timestep logic can violate max_dt

### ISSUE THREAD

**Issue by maxpkatz** [AMReX-Astro/Castro#497]
The logic we use for shrinking the last timestep before stop_time, to ensure that we hit it exactly, can inadvertently have the effect of using a timestep that is larger than max_dt.

**Comment by maxpkatz** [comment]
Example reproducer for 2D KH problem:
```
mpirun -np 1 ./Castro2d.gnu.MPI.OMP.ex inputs.2d max_step=12 stop_time=1.2e-3 castro.max_dt=1.e-4 amr.plot_per=1.e-3 castro.init_shrink=0.99999999 castro.change_max=1.000000001
```
At the last step we will have
```
STEP = 12 TIME = 0.0012 DT = 0.0001000000055
```
where this last dt was (slightly) larger than max_dt.

**Comment by maxpkatz** [comment]
Fixed in c883cd07
